Upgrading to 2.14 breaks REST resources (potentially because of BeanParam)

### Describe the bug

Upgrading from 2.13.4 (everything working/tests run) to 2.14 and have run into an issue it appears with the @BeanParams. This prevents us from upgrading other dependencies such as Stork and Helm as well as being able to take advantage of some nice enhancements in 2.14.

There looks to be two issues but they might be related.

1. Implementing an interface from a REST client in a resource and the method takes in a payload uses @BeanParam throws an exception at build time. 

**[ERROR] Caused by: javax.enterprise.inject.spi.DeploymentException: No annotations found on fields at 'companyx.api.feature.Feature'. Annotations like `@QueryParam` should be used in fields, not in methods.**

```
[ERROR]         [error]: Build step io.quarkus.resteasy.reactive.server.deployment.ResteasyReactiveProcessor#setupEndpoints threw an exception: java.lang.RuntimeException: java.lang.RuntimeException: Failed to process method 'companyx.api.feature.FeatureClient#save'
[ERROR]         at org.jboss.resteasy.reactive.common.processor.EndpointIndexer.createEndpoints(EndpointIndexer.java:318)
[ERROR]         at io.quarkus.resteasy.reactive.server.deployment.ResteasyReactiveProcessor.setupEndpoints(ResteasyReactiveProcessor.java:613)
[ERROR]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[ERROR]         at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR]         at java.base/java.lang.reflect.Method.invoke(Method.java:568)
[ERROR]         at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
[ERROR]         at io.quarkus.builder.BuildContext.run(BuildContext.java:281)
[ERROR]         at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
[ERROR]         at java.base/java.lang.Thread.run(Thread.java:833)
[ERROR]         at org.jboss.threads.JBossThread.run(JBossThread.java:501)
[ERROR] Caused by: java.lang.RuntimeException: Failed to process method 'companyx.api.feature.FeatureClient#save'
[ERROR]         at org.jboss.resteasy.reactive.common.processor.EndpointIndexer.createResourceMethod(EndpointIndexer.java:710)
[ERROR]         at org.jboss.resteasy.reactive.common.processor.EndpointIndexer.createEndpoints(EndpointIndexer.java:408)
[ERROR]         at org.jboss.resteasy.reactive.common.processor.EndpointIndexer.createEndpoints(EndpointIndexer.java:455)
[ERROR]         at org.jboss.resteasy.reactive.common.processor.EndpointIndexer.createEndpoints(EndpointIndexer.java:289)
[ERROR]         ... 12 more
[ERROR] Caused by: javax.enterprise.inject.spi.DeploymentException: No annotations found on fields at 'companyx.api.feature.Feature'. Annotations like `@QueryParam` should be used in fields, not in methods.
[ERROR]         at org.jboss.resteasy.reactive.server.processor.ServerEndpointIndexer.handleBeanParam(ServerEndpointIndexer.java:185)
[ERROR]         at org.jboss.resteasy.reactive.common.processor.EndpointIndexer.createResourceMethod(EndpointIndexer.java:568)
[ERROR]         ... 15 more
```

2. Running with tests, and this is a simple test I get this issue:

**Caused by: javax.enterprise.inject.AmbiguousResolutionException: Beans: [CLASS bean [class=companyx.api.project.ProjectParams, id=9fbe6c147b4e4613d0947934fb37c85f9738382b], CLASS bean [class=companyx.api.storage.AssetParams, id=200da7b2e76e0435661fa5435234030fd029f6f6], CLASS bean [class=companyx.api.artifact.ArtifactParams, id=ccde4a7e2f9460f415624c81466a3936ac3253d7], CLASS bean [class=companyx.api.query.RowParams, id=fef36db73899ee70ae3eed9ab9975fc01852e211]]**

```
[ERROR] companyx.api.artifact.ArtifactParamTest.should_have_all_args  Time elapsed: 0.008 s  <<< ERROR!
java.lang.RuntimeException: java.lang.ExceptionInInitializerError
        at io.quarkus.test.junit.QuarkusTestExtension.throwBootFailureException(QuarkusTestExtension.java:625)
        at io.quarkus.test.junit.QuarkusTestExtension.interceptTestClassConstructor(QuarkusTestExtension.java:696)
        at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.lambda$invoke$0(InterceptingExecutableInvoker.java:93)
        at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
        at org.junit.jupiter.api.extension.InvocationInterceptor.interceptTestClassConstructor(InvocationInterceptor.java:73)
        at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.lambda$invoke$0(InterceptingExecutableInvoker.java:93)
        at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
        at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
        at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
        at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
        at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.invoke(InterceptingExecutableInvoker.java:92)
        at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.invoke(InterceptingExecutableInvoker.java:62)
        at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.invokeTestClassConstructor(ClassBasedTestDescriptor.java:363)
        at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.instantiateTestClass(ClassBasedTestDescriptor.java:310)
        at org.junit.jupiter.engine.descriptor.ClassTestDescriptor.instantiateTestClass(ClassTestDescriptor.java:79)
        at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.instantiateAndPostProcessTestInstance(ClassBasedTestDescriptor.java:286)
        at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$testInstancesProvider$4(ClassBasedTestDescriptor.java:278)
        at java.base/java.util.Optional.orElseGet(Optional.java:364)
        at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$testInstancesProvider$5(ClassBasedTestDescriptor.java:277)
        at org.junit.jupiter.engine.execution.TestInstancesProvider.getTestInstances(TestInstancesProvider.java:31)
        at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$prepare$0(TestMethodTestDescriptor.java:105)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.prepare(TestMethodTestDescriptor.java:104)
        at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.prepare(TestMethodTestDescriptor.java:68)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$prepare$2(NodeTestTask.java:123)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.prepare(NodeTestTask.java:123)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:90)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
        at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
        at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
        at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
        at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
        at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:35)
        at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
        at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:54)
        at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:147)
        at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:127)
        at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:90)
        at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:55)
        at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:102)
        at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:54)
        at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:114)
        at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:86)
        at org.junit.platform.launcher.core.DefaultLauncherSession$DelegatingLauncher.execute(DefaultLauncherSession.java:86)
        at org.apache.maven.surefire.junitplatform.LazyLauncher.execute(LazyLauncher.java:55)
        at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.execute(JUnitPlatformProvider.java:223)
        at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:175)
        at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:139)
        at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:456)
        at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:169)
        at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:595)
        at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:581)
Caused by: java.lang.ExceptionInInitializerError
        at java.base/java.lang.Class.forName0(Native Method)
        at java.base/java.lang.Class.forName(Class.java:467)
        at io.quarkus.runner.bootstrap.StartupActionImpl.run(StartupActionImpl.java:237)
        at io.quarkus.test.junit.QuarkusTestExtension.doJavaStart(QuarkusTestExtension.java:250)
        at io.quarkus.test.junit.QuarkusTestExtension.ensureStarted(QuarkusTestExtension.java:592)
        at io.quarkus.test.junit.QuarkusTestExtension.beforeAll(QuarkusTestExtension.java:640)
        at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$invokeBeforeAllCallbacks$12(ClassBasedTestDescriptor.java:395)
        at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
        at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.invokeBeforeAllCallbacks(ClassBasedTestDescriptor.java:395)
        at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:211)
        at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:84)
        at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:148)
        ... 37 more
Caused by: java.lang.RuntimeException: Failed to start quarkus
        at io.quarkus.runner.ApplicationImpl.<clinit>(Unknown Source)
        ... 49 more
Caused by: javax.enterprise.inject.AmbiguousResolutionException: Beans: [CLASS bean [class=companyx.api.project.ProjectParams, id=9fbe6c147b4e4613d0947934fb37c85f9738382b], CLASS bean [class=companyx.api.storage.AssetParams, id=200da7b2e76e0435661fa5435234030fd029f6f6], CLASS bean [class=companyx.api.artifact.ArtifactParams, id=ccde4a7e2f9460f415624c81466a3936ac3253d7], CLASS bean [class=companyx.api.query.RowParams, id=fef36db73899ee70ae3eed9ab9975fc01852e211]]
        at io.quarkus.arc.impl.ArcContainerImpl.createInstanceSupplier(ArcContainerImpl.java:268)
        at io.quarkus.arc.impl.ArcContainerImpl.beanInstanceSupplier(ArcContainerImpl.java:240)
        at io.quarkus.arc.runtime.BeanContainerImpl.beanInstanceFactory(BeanContainerImpl.java:26)
        at io.quarkus.resteasy.reactive.common.runtime.ArcBeanFactory.<init>(ArcBeanFactory.java:15)
        at io.quarkus.resteasy.reactive.server.runtime.ResteasyReactiveRecorder$11.apply(ResteasyReactiveRecorder.java:311)
        at io.quarkus.resteasy.reactive.server.runtime.ResteasyReactiveRecorder$11.apply(ResteasyReactiveRecorder.java:308)
        at org.jboss.resteasy.reactive.server.core.startup.RuntimeResourceDeployment.parameterExtractor(RuntimeResourceDeployment.java:688)
        at org.jboss.resteasy.reactive.server.core.startup.RuntimeResourceDeployment.buildResourceMethod(RuntimeResourceDeployment.java:341)
        at org.jboss.resteasy.reactive.server.core.startup.RuntimeDeploymentManager.deploy(RuntimeDeploymentManager.java:118)
        at io.quarkus.resteasy.reactive.server.runtime.ResteasyReactiveRecorder.createDeployment(ResteasyReactiveRecorder.java:180)
        at io.quarkus.deployment.steps.ResteasyReactiveProcessor$setupDeployment208273900.deploy_1(Unknown Source)
        at io.quarkus.deployment.steps.ResteasyReactiveProcessor$setupDeployment208273900.deploy(Unknown Source)
        ... 50 more
```


### Expected behavior

2.13 worked.  These are generally pretty simple classes used for @BeanParams in both RESTful endpoints and the REST client.

### Actual behavior

Fails to build and register REST endpoints and give ambiguous errors on classes used as @BeanParam.

**Running build only**

[ERROR] Caused by: javax.enterprise.inject.spi.DeploymentException: No annotations found on fields at 'companyx.api.feature.Feature'. Annotations like `@QueryParam` should be used in fields, not in methods.

**Running with tests**

Caused by: javax.enterprise.inject.AmbiguousResolutionException: Beans: [CLASS bean [class=companyx.api.project.ProjectParams, id=9fbe6c147b4e4613d0947934fb37c85f9738382b], CLASS bean [class=companyx.api.storage.AssetParams, id=200da7b2e76e0435661fa5435234030fd029f6f6], CLASS bean [class=companyx.api.artifact.ArtifactParams, id=ccde4a7e2f9460f415624c81466a3936ac3253d7], CLASS bean [class=companyx.api.query.RowParams, id=fef36db73899ee70ae3eed9ab9975fc01852e211]]




### How to Reproduce?

 _For the life of me, I cannot seem to replicate this but I am trying_.

Scenario 1 - Just build the codebase (multimodule project) without tests.

Scenario 2 - Run the tests.

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

17.0.2

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.14.0.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Maven 3.8.4

### Additional information

Similar to:
- https://github.com/quarkusio/quarkus/issues/22941

What is interesting, is this is the first failure in a test that almost does nothing...

```java
@Disabled
@QuarkusTest
@Tag("functional")
public class ArtifactParamTest {

    @Test
    void should_have_all_args() {
        ArtifactParams params = new ArtifactParams();
        params.setClient("a");
        params.setProject("b");
        params.setArtifact("c");
        assertNotNull(params.getClient());
        assertNotNull(params.getProject());
        assertNotNull(params.getArtifact());
    }
```

Giving this error:

```
Caused by: javax.enterprise.inject.AmbiguousResolutionException: Beans: [CLASS bean [class=companyx.api.artifact.ArtifactParams, id=ccde4a7e2f9460f415624c81466a3936ac3253d7], CLASS bean [class=companyx.api.storage.AssetParams, id=200da7b2e76e0435661fa5435234030fd029f6f6], CLASS bean [class=companyx.api.project.ProjectParams, id=9fbe6c147b4e4613d0947934fb37c85f9738382b], CLASS bean [class=companyx.api.query.RowParams, id=fef36db73899ee70ae3eed9ab9975fc01852e211]]
```

These are the classes:

```java
@Data
@NoArgsConstructor
@EqualsAndHashCode(callSuper = true)
public class ArtifactParams extends ProjectParams {

    @PathParam("artifact")
    private String artifact;

    public ArtifactParams(final String artifact, final String project, final String client) {
        super(project, client);
        this.artifact = artifact;
    }
}
```

```java
@Data
@NoArgsConstructor
@EqualsAndHashCode(callSuper = true)
public class ProjectParams extends ClientParams {

    @PathParam("project")
    private String project;

    public ProjectParams(final String project, final String client) {
        super(client);
        this.project = project;
    }
}
```

```java
@Data
public class AssetParams extends ProjectParams {
    
    @PathParam("filename")
    private String filename;

    @PathParam("ext")
    private String ext;
    
    @JsonIgnore
    public final String getFilename() {
        return filename + '.' + ext;
    }
}
```

I may of found one of the issues when "just building" - have `@BeanParam` at all was throwing everything into a tizzy.  Removing it everywhere allows me to get past this issue of ambiguous beans.

```
[ERROR] Caused by: javax.enterprise.inject.spi.DeploymentException: No annotations found on fields at 'companyx.api.feature.Feature'. Annotations like @QueryParam should be used in fields, not in methods.
```

I think I found it, or at least the cause.  Once we got past the ambiguous bean compilation error it seem to have nothing more to do with that.

What I realized was that only methods referencing a specific bean was causing the problems.  Didn't matter if it came from a RESTClient, had a @BeanParam or not, just annotating it as a REST method, Lombok or not - they all failed.

The failure was the same in all cases and specific to the a single payload.  If I took that payload out everything worked.

```java
quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR]         [error]: Build step io.quarkus.resteasy.reactive.server.deployment.ResteasyReactiveProcessor#setupEndpoints threw an exception: java.lang.RuntimeException: java.lang.RuntimeException: Failed to process method 'companyx.api.extension.ExtensionClient#save'
[ERROR]         at org.jboss.resteasy.reactive.common.processor.EndpointIndexer.createEndpoints(EndpointIndexer.java:315)
[ERROR]         at io.quarkus.resteasy.reactive.server.deployment.ResteasyReactiveProcessor.setupEndpoints(ResteasyReactiveProcessor.java:613)
[ERROR]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[ERROR]         at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR]         at java.base/java.lang.reflect.Method.invoke(Method.java:568)
[ERROR]         at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
[ERROR]         at io.quarkus.builder.BuildContext.run(BuildContext.java:281)
[ERROR]         at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
[ERROR]         at java.base/java.lang.Thread.run(Thread.java:833)
[ERROR]         at org.jboss.threads.JBossThread.run(JBossThread.java:501)
[ERROR] Caused by: java.lang.RuntimeException: Failed to process method 'companyx.api.extension.ExtensionClient#save'
[ERROR]         at org.jboss.resteasy.reactive.common.processor.EndpointIndexer.createResourceMethod(EndpointIndexer.java:754)
[ERROR]         at org.jboss.resteasy.reactive.common.processor.EndpointIndexer.createEndpoints(EndpointIndexer.java:404)
[ERROR]         at org.jboss.resteasy.reactive.common.processor.EndpointIndexer.createEndpoints(EndpointIndexer.java:286)
[ERROR]         ... 12 more
[ERROR] Caused by: javax.enterprise.inject.spi.DeploymentException: No annotations found on fields at 'companyx.api.extension.Extension'. Annotations like `@QueryParam` should be used in fields, not in methods.
[ERROR]         at org.jboss.resteasy.reactive.server.processor.ServerEndpointIndexer.handleBeanParam(ServerEndpointIndexer.java:185)
[ERROR]         at org.jboss.resteasy.reactive.common.processor.EndpointIndexer.createResourceMethod(EndpointIndexer.java:612)
[ERROR]         ... 14 more
[ERROR] -> [Help 1]
```

So, I took that specific bean apart piece by piece until I could reproduce the error again...ithe error occurs when it hits this annotation `@DefaultValue("created")`.  If I take it out it works fine.

```java
@Schema(hidden = true)
@DefaultValue("CREATED")
private ExtensionStatus status = ExtensionStatus.CREATED;
```

I believe the reason is that by ussing that @DefaultValue we triggers the @BeanParam (per the javadocs of the annotation) but it is not a BeanParam!  That is why the error about nothing being annotated correctly gets thrown because it thought that payload was a BeanParam.

This is also why we could not reproduce it (or if we did it was by mistake and not obvious).  Never thought about having all the annotated fields in there...

So, you fixed the ambiguous BeanParam.  This issue was a mis-used annotation.

I think this will close this ticket out once your changes are merged in!  Thanks for all your help.

--- a/extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/java/io/quarkus/resteasy/reactive/server/deployment/ResteasyReactiveProcessor.java
+++ b/extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/java/io/quarkus/resteasy/reactive/server/deployment/ResteasyReactiveProcessor.java
@@ -730,7 +730,7 @@ public class ResteasyReactiveProcessor {
         return ab.get();
     }
 
-    // We want to add @Typed to resources and providers so that they can be resolved as CDI bean using purely their
+    // We want to add @Typed to resources, beanparams and providers so that they can be resolved as CDI bean using purely their
     // class as a bean type. This removes any ambiguity that potential subclasses may have.
     @BuildStep
     public void transformEndpoints(
@@ -743,6 +743,10 @@ public class ResteasyReactiveProcessor {
         Set<DotName> allResources = new HashSet<>();
         allResources.addAll(resourceScanningResultBuildItem.getResult().getScannedResources().keySet());
         allResources.addAll(resourceScanningResultBuildItem.getResult().getPossibleSubResources().keySet());
+
+        // all found bean params
+        Set<String> beanParams = resourceScanningResultBuildItem.getResult()
+                .getBeanParams();
 
         // discovered filters and interceptors
         Set<String> filtersAndInterceptors = new HashSet<>();
@@ -788,6 +792,14 @@ public class ResteasyReactiveProcessor {
                                 && clazz.declaredAnnotation(ResteasyReactiveDotNames.TYPED) == null) {
                             // Add @Typed(MyResource.class)
                             context.transform().add(createTypedAnnotationInstance(clazz, beanArchiveIndexBuildItem)).done();
+                            return;
+                        }
+                        // check if the class is a bean param
+                        if (beanParams.contains(clazz.name().toString())
+                                && clazz.declaredAnnotation(ResteasyReactiveDotNames.TYPED) == null) {
+                            // Add @Typed(MyBean.class)
+                            context.transform().add(createTypedAnnotationInstance(clazz, beanArchiveIndexBuildItem)).done();
+                            return;
                         }
                     }
                 }));

--- a/extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/test/java/io/quarkus/resteasy/reactive/server/test/beanparam/BeanParamTest.java
+++ b/extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/test/java/io/quarkus/resteasy/reactive/server/test/beanparam/BeanParamTest.java
@@ -1,0 +1,104 @@
+package io.quarkus.resteasy.reactive.server.test.beanparam;
+
+import javax.ws.rs.BeanParam;
+import javax.ws.rs.CookieParam;
+import javax.ws.rs.FormParam;
+import javax.ws.rs.HeaderParam;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.QueryParam;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.test.QuarkusUnitTest;
+
+public class BeanParamTest {
+    @RegisterExtension
+    static final QuarkusUnitTest TEST = new QuarkusUnitTest()
+            .setArchiveProducer(() -> {
+                return ShrinkWrap.create(JavaArchive.class)
+                        .addClasses(MyBeanParamWithFieldsAndProperties.class, Top.class);
+            });
+
+    @Test
+    void shouldDeployWithoutIssues() {
+        // we only need to check that it deploys
+    }
+
+    public static class Top {
+        @PathParam("pathParam")
+        private String pathParam = "pathParam";
+
+        public String getPathParam() {
+            return pathParam;
+        }
+
+        public void setPathParam(String pathParam) {
+            this.pathParam = pathParam;
+        }
+    }
+
+    public static class MyBeanParamWithFieldsAndProperties extends Top {
+        @HeaderParam("headerParam")
+        private String headerParam = "headerParam";
+        @CookieParam("cookieParam")
+        private String cookieParam = "cookieParam";
+        @FormParam("formParam")
+        private String formParam = "formParam";
+        @QueryParam("queryParam")
+        private String queryParam = "queryParam";
+
+        // FIXME: Matrix not supported
+
+        public String getHeaderParam() {
+            return headerParam;
+        }
+
+        public void setHeaderParam(String headerParam) {
+            this.headerParam = headerParam;
+        }
+
+        public String getCookieParam() {
+            return cookieParam;
+        }
+
+        public void setCookieParam(String cookieParam) {
+            this.cookieParam = cookieParam;
+        }
+
+        public String getFormParam() {
+            return formParam;
+        }
+
+        public void setFormParam(String formParam) {
+            this.formParam = formParam;
+        }
+
+        public String getQueryParam() {
+            return queryParam;
+        }
+
+        public void setQueryParam(String queryParam) {
+            this.queryParam = queryParam;
+        }
+    }
+
+    @Path("/")
+    public static class Resource {
+        @Path("/a/{restPathDefault}/{restPath_Overridden}/{pathParam}")
+        @POST
+        public String beanParamWithFields(@BeanParam MyBeanParamWithFieldsAndProperties p) {
+            return null;
+        }
+
+        @Path("/b/{pathParam}")
+        @POST
+        public String beanParamWithFields(@BeanParam Top p) {
+            return null;
+        }
+    }
+}

